internal/sys/fd_trace.go

```diff
@@ -0,0 +1,137 @@
+package sys
+
+import (
+	"bytes"
+	"fmt"
+	"runtime"
+	"strings"
+	"sync"
+)
+
+// FDs is a registry of all file descriptors wrapped into sys.FDs that were
+// created while an fd tracer was active.
+var FDs = newFDMap()
+
+// Finalize is called when the Go runtime frees an [FD].
+var Finalize FDTracer
+
+// traceFDs returns true if there are active fd tracers registered to the
+// package.
+func traceFDs() bool {
+	return Finalize != nil
```

Comment:
Doesn't seem worth it to have a function for a single boolean?

Reply:
There used to be multiple hooks, so this made sense. `if traceFDs() { ... }` is still a bit more expressive than `if TraceLeakedFD != nil { ... }`, but ok, I'll remove it.

Reply:
Keep it if you prefer it.

Reply:
Ended up removing it since `FD.finalize()` needed a nil check on `onLeakFD`.